File: output/gemma4-e2b/avatar/Python/Java/codeforces_621_A.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class codeforces_621_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read n
        if (!scanner.hasNextInt()) return;
        int n = scanner.nextInt();

        // Read the list elements
        List<Integer> lst = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            if (scanner.hasNextInt()) {
                lst.add(scanner.nextInt());
            } else {
                // Handle case where input might be incomplete, though usually guaranteed in these problems
                break;
            }
        }

        // lst.sort()
        Collections.sort(lst);

        // lst.reverse()
        Collections.reverse(lst);

        // Calculate od (count of odd numbers)
        int od = 0;
        for (int i : lst) {
            if ((i & 1) != 0) {
                od++;
            }
        }

        // Calculate sum
        int sum = 0;
        boolean ok = false; // Corresponds to ok = 0 (False)

        // Loop based on n
        for (int i = 0; i < n; i++) {
            int currentElement = lst.get(i);
            
            if (currentElement % 2 == 0) {
                sum += currentElement;
            } else {
                // Python: if ( ok or od > 1 ):
                if (ok || od > 1) {
                    sum += currentElement;
                    od -= 1;
                    ok = !ok; // ok = not ok
                }
            }
        }

        // Print sum
        System.out.println(sum);
        
        scanner.close();
    }
}

File: output/gemma4-e2b/avatar/Java/Python/codeforces_621_A.py
import sys

def solve():
    # Read all input tokens
    data = sys.stdin.read().split()
    
    if not data:
        return

    # The first token is N
    try:
        n = int(data[0])
    except ValueError:
        return # Handle case where N is not an integer

    if len(data) < n + 1:
        # Not enough numbers provided
        return

    # The remaining tokens are the list of numbers
    list_of_numbers = [int(x) for x in data[1:n+1]]

    list_odd = []
    sum_even = 0

    # Separate even and odd numbers
    for x in list_of_numbers:
        if x % 2 == 0:
            sum_even += x
        else:
            list_odd.append(x)

    # Sort the odd numbers
    list_odd.sort()

    # Calculate the total sum (sum of evens + sum of odds)
    total_sum = sum_even
    for x in list_odd:
        total_sum += x

    # Apply the final adjustment
    # If the count of odd numbers is odd, subtract the smallest one
    if len(list_odd) % 2 != 0:
        total_sum -= list_odd[0]

    print(total_sum)

solve()